File: go/isogram/isogram.go
package isogram

import "unicode"

func IsIsogram(word string) bool {
	if word == "" {
		return true
	}

	var seen uint32

	for _, r := range word {
		if !unicode.IsLetter(r) {
			continue
		}

		r = unicode.ToLower(r)
		if r < 'a' || r > 'z' {
			return false
		}

		bit := uint32(1) << (r - 'a')
		if seen&bit != 0 {
			return false
		}
		seen |= bit
	}

	return true
}

File: ruby/isogram/isogram.rb
module Isogram
  def self.isogram?(input)
    return false if input.nil?

    seen = 0

    input.each_char do |ch|
      next unless ch.match?(/[A-Za-z]/)

      bit = (ch.downcase.ord - "a".ord)
      return false if bit < 0 || bit > 25

      mask = 1 << bit
      return false if (seen & mask) != 0

      seen |= mask
    end
  end
end
